Unsupported engine due to @itwin/core-telemetry dependency

Different versions of itwin core packages are installed with different supported node versions due to `@itwin/core-telemetry`.
Consider moving `@itwin/core-telemetry` dependency of `@itwin/appui-react` to **peerDependencies**.

I.e. when [engine-strict](https://docs.npmjs.com/cli/v9/using-npm/config#engine-strict) is enabled:
```
npm i @itwin/appui-react@4.0.0-dev.15 --engine-strict=true
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: @itwin/cloud-agnostic-core@1.4.0
npm ERR! notsup Not compatible with your version of node/npm: @itwin/cloud-agnostic-core@1.4.0
npm ERR! notsup Required: {"node":">=12.20 <17.0.0"}
npm ERR! notsup Actual:   {"npm":"9.5.0","node":"v18.15.0"}
```

Respective error using pnpm:
```
ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

This error happened while installing the dependencies of @itwin/appui-react@4.0.0-dev.15
at @itwin/core-telemetry@3.7.0
at @itwin/core-common@3.7.0

Your Node version is incompatible with "registry.npmjs.org/@itwin/object-storage-core/1.4.0".

Expected version: >=12.20 <17.0.0
Got: v18.15.0
```

--- a/common/changes/@itwin/appui-react/core-telemetry-peer-dep_2023-04-06-10-26.json
+++ b/common/changes/@itwin/appui-react/core-telemetry-peer-dep_2023-04-06-10-26.json
@@ -1,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/appui-react",
+      "comment": "Move @itwin/core-telemetry to peerDependencies.",
+      "type": "none"
+    }
+  ],
+  "packageName": "@itwin/appui-react"
+}

--- a/ui/appui-react/package.json
+++ b/ui/appui-react/package.json
@@ -53,6 +53,7 @@
     "@itwin/core-geometry": ">=3.7.0",
     "@itwin/core-quantity": ">=3.7.0",
     "@itwin/core-react": "workspace:^4.0.0-dev.15",
+    "@itwin/core-telemetry": ">=3.7.0",
     "@itwin/imodel-components-react": "workspace:^4.0.0-dev.15",
     "react": "^17.0.0 || ^18.0.0",
     "react-dom": "^17.0.0 || ^18.0.0",
@@ -75,6 +76,7 @@
     "@itwin/core-orbitgt": "^3.7.0",
     "@itwin/core-quantity": "^3.7.0",
     "@itwin/core-react": "workspace:*",
+    "@itwin/core-telemetry": "^3.7.0",
     "@itwin/ecschema-metadata": "^3.7.0",
     "@itwin/eslint-plugin": "^3.7.0",
     "@itwin/imodel-components-react": "workspace:*",
@@ -127,7 +129,6 @@
   ],
   "dependencies": {
     "@bentley/icons-generic": "^1.0.34",
-    "@itwin/core-telemetry": "^3.7.0",
     "@itwin/itwinui-react": "^2.5.0",
     "@itwin/itwinui-variables": "^2.0.0",
     "@itwin/itwinui-icons-react": "^2.1.0",

